src/build/env.py

```diff
@@ -141,15 +154,23 @@ def __enter__(self):  # type: () -> IsolatedEnvironment
         if 'PATH' in os.environ:
             exe_path.append(os.environ['PATH'])
 
-        self._replace_env('PATH', os.pathsep.join(exe_path))
-        self._replace_env('PYTHONPATH', os.pathsep.join(sys_path))
-        self._replace_env('PYTHONHOME', self.path)
-        self._pop_env('PIP_REQUIRE_VIRTUALENV')
-
         self._symlink_relative(sysconfig.get_path('include'))
         self._symlink_relative(sysconfig.get_path('platinclude'))
         self._symlink_relative(sysconfig.get_config_var('LIBPL'))
 
+        self._symlink_file(sys.executable, os.path.join('bin', 'python'))
+        exe_path.insert(0, os.path.join(self.path, 'bin'))
+
+        self._executable = sys.executable
+        sys.executable = os.path.join(self.path, 'bin', 'python')
```

Comment:
I'm not in love with overwritting the sys.executable. Should detail with a comment here why.

Reply:
Me neither :confused:. I will add a comment.